net/portmapper/portmapper.go

```diff
@@ -803,12 +811,59 @@ func (c *Client) Probe(ctx context.Context) (res ProbeResult, err error) {
 		uc.WriteToUDPAddrPort(uPnPIGDPacket, upnpMulticastAddr)
 	}
 
+	// We can see multiple UPnP responses from LANs with multiple
+	// UPnP-capable routers. Rather than randomly picking whichever arrives
+	// first, let's collect all UPnP responses and choose at the end.
+	//
+	// We do this by starting a 50ms timer from when the first UDP packet
+	// is received, and waiting at least that long for more UPnP responses
+	// to arrive before returning.
```

Comment:
how does the 50ms timer relate to the 250ms bounded context on L736 that is tied to `uc`? Maybe a doc would help.

Reply:
Just pushed a change with more comments; PTAL?

Reply:
We wait at least 50ms before returning, so long as the first packet is seen within the first 200ms of the context creation, which is likely in the common case IIUC.